share/spack/qa/run-docker-tests

```diff
@@ -21,9 +21,18 @@ ensure_docker_login() {
         return $__login_success
     fi
 
+    # NOTE: work around an issue with docker/docker hub
+    # https://github.com/docker/hub-feedback/issues/1222
+    # https://github.com/docker/cli/issues/1180
+    rm -f $HOME/.docker/config.json
```

Comment:
this is definitely only run on CI and never as a developer user running tests in his/her userspace?

Reply:
Good point.  It didn't occur to me that the `run-XYZ-tests` scripts were for anything other than CI.  What do you suggest?   Should I try to detect a travis environment?  Perhaps this whole workaround is problematic?

Reply:
If it only occurs on travis, we could maybe place this work-around higher up to where it is called.

Otherwise, maybe guard it with `"${CI}" == "true"` or similar? https://docs.travis-ci.com/user/environment-variables/#default-environment-variables